src/app/AttributePathParams.h

```diff
@@ -81,10 +83,49 @@ struct AttributePathParams
         return true;
     }
 
-    ClusterId mClusterId     = kInvalidClusterId;   // uint32
-    AttributeId mAttributeId = kInvalidAttributeId; // uint32
-    EndpointId mEndpointId   = kInvalidEndpointId;  // uint16
-    ListIndex mListIndex     = kInvalidListIndex;   // uint16
+    bool IsAttributePathIntersect(const DataVersionFilter & other) const
+    {
+        VerifyOrReturnError(HasWildcardEndpointId() || mEndpointId == other.mEndpointId, false);
+        VerifyOrReturnError(HasWildcardClusterId() || mClusterId == other.mClusterId, false);
+
+        return true;
+    }
+
+    bool IsAttributePathIntersect(const ConcreteClusterPath & other) const
+    {
+        VerifyOrReturnError(HasWildcardEndpointId() || mEndpointId == other.mEndpointId, false);
+        VerifyOrReturnError(HasWildcardClusterId() || mClusterId == other.mClusterId, false);
+
+        return true;
+    }
+
+    bool IsWildcardRequest(const AttributePathParams & aOther) const
+    {
+        if (HasWildcardEndpointId())
+        {
+            return true;
+        }
+        if (HasWildcardClusterId())
+        {
+            return true;
+        }
+        if (mEndpointId == aOther.mEndpointId && mClusterId == aOther.mClusterId && HasWildcardAttributeId())
+        {
+            return true;
+        }
+        return false;
+    }
+
+    bool operator<(const AttributePathParams & aOther) const
+    {
+        return mpReadClient < aOther.mpReadClient || mEndpointId < aOther.mEndpointId || mClusterId < aOther.mClusterId;
+    }
+
+    ClusterId mClusterId      = kInvalidClusterId;   // uint32
+    AttributeId mAttributeId  = kInvalidAttributeId; // uint32
+    EndpointId mEndpointId    = kInvalidEndpointId;  // uint16
+    ListIndex mListIndex      = kInvalidListIndex;   // uint16
+    ReadClient * mpReadClient = nullptr;
```

Comment:
mpReadClient should not be needed if I'm right about when we can clear our paths in the cache....

Reply:
for instance, first subscribe with C1, then read using same cache for C1, we need mpReadClient to avoid the removal of read request paths from first subscribe when second read closes

Reply:
We never discussed needing this. The algorithm as agreed to doesn't require this scoping to a read client instance.

Reply:
I think our discussed algorithm is to clear up read request path in cache when onDone is triggered. If ReadClient 1 subscribe paths, then ReadClient read paths using same cache, the onDone in ReadClient 2 would clear out the read request paths for ReadClient 1, which would be problematic

Reply:
removed, thanks